pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -209,6 +209,11 @@ type PipelineTask struct {
 	Timeout *metav1.Duration `json:"timeout,omitempty"`
 }
 
+// IsCustomTask checks whether an embedded TaskSpec is a Custom Task
+func (et *EmbeddedTask) IsCustomTask() bool {
+	return et != nil && et.APIVersion != "" && et.Kind != ""
```

Comment:
2 questions:
- isn't it technically possible to use `kind` "Task"?
- Should this depend on whether `et` has a `spec` vs a `taskspec`?

Reply:
Yes, it is possible to have kind "Task" in a Custom Task in the current implementation - and this PR does not change that

Reply:
re: don't want to change the behavior to check for spec vs taskspec, want this to be a simple refactor then other fixes can be done on top of it